src/solidity-frontend/solidity_convert.cpp

```diff
@@ -5182,6 +5233,54 @@ bool solidity_convertert::get_expr(
       new_expr = call;
       break;
     }
+    // * special: new D{value: amount}(4), handle the {value: amount} part
```

Comment:
Why do we handle new expressions in CallExprClass? Maybe we can have a `NewOptionExpr`